### PR TITLE
Expand analysis and visualization tools

### PR DESCRIPTION
**Pull request by andey-robins**
This PR does a handful of things:

1. Adds in some automated experimental data collection. Run `make experiment` in the `src/` directory to generate data. Note, I didn't extend the java code to create all the intermediary folders, so there may be some manual setup required.
2. Updated the gitignore to support the python and data dependencies now collected.
3. Fixed a couple bugs in the xy-cartesian representation
    - Capped values within the unit circle
    - Added coordinate output
    - Normalized between [-1,1]
4. Adds gif generator to convert a random run into a generation-by-generation animation of the best scattering
5. Adds python dependency management requirements file and freezes pip dependencies

### CODE REVIEW

**Comment by andey-robins** [comment]
@christianzski or @JavaNoTea I know you probably can't approve the merge with the button, but I still wanna get sign off from one of ya before merging this in.

**Review by christianzski — APPROVED** [review]
@andey-robins Looks good to me, nice job! Gifs are pretty sweet too.

**Pull request merged**
